### PR TITLE
Fix Client Table

### PR DESCRIPTION
**Pull request by richardliaw**
Hides append-only log in the Python implementation.

Should partially address #2852.

TODO:
 - [x] Add tests for this

### CODE REVIEW

**Comment by richardliaw** [comment]
OK I added a light test for this using subprocess to start ray; this may only work in xray? and may also require `https://pypi.org/project/pytest-timeout/` to be installed on Travis.

Is there a better way to achieve what I'm trying to do?

File: python/ray/test/test_global_state.py [hunk 1]
```diff
@@ -3,56 +3,114 @@
...
-    ray.get(cpu_task.remote(0))
+REDIS_PORT = 6543
 
 
 @ray.remote(num_cpus=1)
 def cpu_task(seconds):
     time.sleep(seconds)
 
 
+def _get_raylet_pid(raylet_socket):
```
Comment (richardliaw) [inline python/ray/test/test_global_state.py]: Hey @robertnishihara, is there a better way of getting the Raylet PID?

**Review by robertnishihara — APPROVED** [review]
Looks good to me, pending tests passing.

**Comment by robertnishihara** [comment]
jenkins, retest this please

**Pull request merged**